public/app/features/integrated-alerting/components/NotificationChannel/AddNotificationChannelModal/AddNotificationChannelModal.tsx

```diff
@@ -11,18 +11,27 @@ import { getStyles } from './AddNotificationChannelModal.styles';
 import { Messages } from './AddNotificationChannelModal.messages';
 import { TYPE_OPTIONS, TYPE_FIELDS_COMPONENT } from './AddNotificationChannel.constants';
 import { NotificationChannelService } from '../NotificationChannel.service';
+import { getInitialValues } from './AddNotificationChannelModal.utils';
 
 const { required } = validators;
-const initialValues = { type: TYPE_OPTIONS[0] };
 
-export const AddNotificationChannelModal: FC<AddNotificationChannelModalProps> = ({ isVisible, setVisible }) => {
+export const AddNotificationChannelModal: FC<AddNotificationChannelModalProps> = ({
+  isVisible,
+  notificationChannel,
+  setVisible,
+}) => {
   const styles = useStyles(getStyles);
+  const initialValues = getInitialValues(notificationChannel);
   const { getNotificationChannels } = useContext(NotificationChannelProvider);
   const onSubmit = async (values: NotificationChannelRenderProps) => {
     try {
-      await NotificationChannelService.add(values);
+      if (notificationChannel) {
+        await NotificationChannelService.change(values);
+      } else {
+        await NotificationChannelService.add(values);
+      }
       setVisible(false);
```

Comment:
I guess `setVisible` would be better put into the finally `section`.

Reply:
Same as the above!